pythran/pythonic/include/types/tuple.hpp

```diff
@@ -1115,23 +1115,32 @@ namespace sutils
   using push_front_t = concat_t<types::pshape<T>, P>;
 
   template <class S>
-  long find(S &s, long v, std::integral_constant<size_t, 0>)
+  long find(S &s, long v, std::integral_constant<size_t, 0>, long start,
+            std::function<bool(long, long)> comp)
   {
-    return v == std::get<0>(s) ? 0 : -1;
+    return comp(std::get<0>(s), v) && 0 < start ? 0 : -1;
   }
   template <class S, size_t I>
-  long find(S &s, long v, std::integral_constant<size_t, I>)
+  long find(S &s, long v, std::integral_constant<size_t, I>, long start,
+            std::function<bool(long, long)> comp)
   {
-    return v == std::get<I>(s)
+    return comp(std::get<I>(s), v) && I < start
                ? I
-               : find(s, v, std::integral_constant<size_t, I - 1>());
+               : find(s, v, std::integral_constant<size_t, I - 1>(), start,
+                      comp);
   }
 
   template <class S>
-  long find(S &s, long v)
+  long
+  find(S &s, long v,
+       long start = std::integral_constant<size_t, std::tuple_size<S>::value>(),
```

Comment:
shouldn't statr default to `0` ?

Reply:
no because you go backward in the template expansion

Reply:
(i.e. you start with len(tuple)-1)

Reply:
And no need to use integral_constant here.

Reply:
Actually yes, because you need the integral_constant constant to finish the expansion (you have two overlads of find(), one for I=0 and one for I != 0, this has to be known at compile time). Or at least that's what I understand. (remove integral_constant breaks compilation by the way)